salt/modules/file.py

```diff
@@ -2093,14 +2093,10 @@ def replace(path,
                         # Content was found, so set found.
                         found = True
 
-                # Keep track of show_changes here, in case the file isn't
-                # modified
-                if show_changes or append_if_not_found or \
-                   prepend_if_not_found:
-                    orig_file = r_data.read(filesize).splitlines(True) \
-                        if isinstance(r_data, mmap.mmap) \
-                        else r_data.splitlines(True)
-                    new_file = result.splitlines(True)
+                orig_file = r_data.read(filesize).splitlines(True) \
```

Comment:
All I am suggesting is that you find another approach to fixing your issue that keeps the gate in place, so that you do not break the use case for those that set `show_changes` to `false` to avoid reading the file into memory twice.